### PR TITLE
fix/3: clean up code

### PR DESCRIPTION
**Pull request by NovaHe**

### CODE REVIEW

File: leetcode/0003.Longest-Substring-Without-Repeating-Characters/3. Longest Substring Without Repeating Characters.go [hunk 1]
```diff
@@ -26,25 +26,35 @@ func lengthOfLongestSubstring(s string) int {
...
 }
 
-// 解法二 滑动窗口
-func lengthOfLongestSubstring_(s string) int {
-	if len(s) == 0 {
-		return 0
+// 解法二 滑动窗口-数组桶
+func lengthOfLongestSubstring1(s string) int {
+	right, left, res := 0, 0, 0
+	var m [256]int
```
Comment (halfrost) [inline leetcode/0003.Longest-Substring-Without-Repeating-Characters/3. Longest Substring Without Repeating Characters.go:32]: Is it better to name variable m to freq?

File: leetcode/0003.Longest-Substring-Without-Repeating-Characters/3. Longest Substring Without Repeating Characters.go [hunk 2]
```diff
@@ -26,25 +26,35 @@ func lengthOfLongestSubstring(s string) int {
...
 	}
-	var freq [256]int
-	result, left, right := 0, 0, -1
+	return res
+}
 
+// 解法二 滑动窗口-哈希桶
+func lengthOfLongestSubstring2(s string) int {
+	right, left, res := 0, 0, 0
+	m := make(map[byte]int, len(s))
```
Comment (halfrost) [inline leetcode/0003.Longest-Substring-Without-Repeating-Characters/3. Longest Substring Without Repeating Characters.go:48]: Is it better to name variable m to indexes?

Comment (NovaHe) [reply leetcode/0003.Longest-Substring-Without-Repeating-Characters/3. Longest Substring Without Repeating Characters.go]: done

File: leetcode/0003.Longest-Substring-Without-Repeating-Characters/3. Longest Substring Without Repeating Characters.go (hunk 1 above)
Comment (NovaHe) [reply leetcode/0003.Longest-Substring-Without-Repeating-Characters/3. Longest Substring Without Repeating Characters.go]: array's value is indexs, rename variable `m` to `indexes`

File: leetcode/0003.Longest-Substring-Without-Repeating-Characters/3. Longest Substring Without Repeating Characters.go [hunk 3]
```diff
@@ -26,25 +26,35 @@ func lengthOfLongestSubstring(s string) int {
 	return result
 }
 
-// 解法二 滑动窗口
-func lengthOfLongestSubstring_(s string) int {
-	if len(s) == 0 {
-		return 0
+// 解法二 滑动窗口-数组桶
+func lengthOfLongestSubstring1(s string) int {
```
Comment (halfrost) [inline leetcode/0003.Longest-Substring-Without-Repeating-Characters/3. Longest Substring Without Repeating Characters.go:30]: This solution is the wrong answer. If s = " ", this solution output 0, but actually expected 1.

**Pull request merged**